### PR TITLE
Remove SiteCreation feature flag and kill dead code

### PR DESCRIPTION
**Pull request by elibud**
**To test:**

Make sure all scenarios for site creation work as expected.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/BlogListViewController.m [hunk 1]
```diff
@@ -12,7 +12,6 @@
 #import "UILabel+SuggestSize.h"
 #import "WordPress-Swift.h"
 #import "WPGUIConstants.h"
-#import "CreateNewBlogViewController.h"
```
Comment (ScoutHarris) [inline WordPress/Classes/ViewRelated/Blog/BlogListViewController.m]: These can be removed as well:

`WPNoResultsViewDelegate, // To be removed with FeatureFlagSiteCreation.`

`@property (nonatomic, strong) WPNoResultsView *noResultsView; // To be removed with FeatureFlagSiteCreation.`

```
// To be removed with FeatureFlagSiteCreation.
- (void)configureNoResultsView
```

```
// To be removed with FeatureFlagSiteCreation.
#pragma mark - WPNoResultsViewDelegate
```

**Comment by elibud** [comment]
Ready for another pass, thank you!

**Comment by ScoutHarris** [comment]
:shipit: !!!

**Comment by diegoreymendez** [comment]
@elibud - I'm a bit lost from the short description in this PR for our release notes.  Can you let me know if this is important and needs to be added?

Thanks!

**Pull request merged**